be/src/runtime/current_thread.h

```diff
@@ -51,6 +57,7 @@ class TUniqueId;
 inline thread_local MemTracker* tls_mem_tracker = nullptr;
 inline thread_local MemTracker* tls_operator_mem_tracker = nullptr;
 inline thread_local MemTracker* tls_exceed_mem_tracker = nullptr;
+inline thread_local MemTracker* tls_singleton_check_mem_tracker = nullptr;
```

Comment:
I add `tls_singleton_check_mem_tracker` because we can't support trace alloc and free object size on different thread (because of statistical errors in object size), so we need to seperate mem tracker and limit check tracker. 

Use `SCOPED_THREAD_LOCAL_SINGLETON_CHECK_MEM_TRACKER_SETTER` to set up a separate check tracker.